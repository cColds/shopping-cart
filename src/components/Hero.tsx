import { Link } from "react-router-dom";

export default function Hero() {
  return (
    <div className="flex max-w-4xl flex-col items-center gap-5 py-12">
      <h1 className="text-center">
        Your one-stop destination for all your shopping needs
      </h1>
      <p className="text-center text-lg">
        Discover an unparalleled shopping experience with our extensive
        selection of products, unbeatable prices, and exceptional customer
        service. Shop now and transform your shopping journey with us.
      </p>
      <Link
        to="/store"
        className="w-[150px] overflow-hidden truncate rounded-lg bg-primary-color p-3 text-center font-bold text-white"
      >
        Shop now
      </Link>
      <p>Hey</p>
    </div>
  );
}
