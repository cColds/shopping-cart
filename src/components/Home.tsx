export default function Home() {
  return (
    <div>
      <h1>Your one-stop destination for all your shopping needs</h1>
    </div>
  );
}
