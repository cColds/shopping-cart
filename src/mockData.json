{
  "data": {
    "products": {
      "edges": [
        {
          "node": {
            "id": "gid://shopify/Product/7982853619734",
            "title": "Slides",
            "description": "Simple, minimal and comfortable, these slides feature a classic design in the perfect shade of iron. Whether you're just lounging around the house or running errands, these slides will offer all-day comfort.",
            "featuredImage": {
              "id": "gid://shopify/ProductImage/39774608687126",
              "url": "https://cdn.shopify.com/s/files/1/0688/1755/1382/products/slides.jpg?v=1675447358"
            },
            "variants": {
              "edges": [
                {
                  "node": {
                    "price": {
                      "amount": "25.0",
                      "currencyCode": "CAD"
                    }
                  }
                }
              ]
            }
          }
        },
        {
          "node": {
            "id": "gid://shopify/Product/7982856273942",
            "title": "Sweatpants",
            "description": "Soft and comfortable sweatpants in stylish shades. They are perfect for lounging with their cozy stretch fabric that offers just the right amount of warmth. Enjoy the ultimate relaxation experience!",
            "featuredImage": {
              "id": "gid://shopify/ProductImage/39774602002454",
              "url": "https://cdn.shopify.com/s/files/1/0688/1755/1382/products/GreenSweatpants01.jpg?v=1675455387"
            },
            "variants": {
              "edges": [
                {
                  "node": {
                    "price": {
                      "amount": "35.0",
                      "currencyCode": "CAD"
                    }
                  }
                }
              ]
            }
          }
        }
      ]
    }
  }
}
