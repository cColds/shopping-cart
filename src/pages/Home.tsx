import FeaturedItems from "../components/FeaturedItems";
import Hero from "../components/Hero";
import Item from "../interfaces/Item";

export default function Home({
  items,
  handleItemClick,
}: {
  items: Item[];
  handleItemClick: (item: Item) => void;
}) {
  return (
    <>
      <Hero />
      <FeaturedItems items={items} handleItemClick={handleItemClick} />
    </>
  );
}
