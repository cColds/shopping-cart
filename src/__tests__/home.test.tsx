import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Home from "../pages/Home";
import mockData from "../mockData.json";
import Hero from "../components/Hero";

it("shop now link should link to store", () => {
  render(
    <MemoryRouter>
      <Hero />
    </MemoryRouter>
  );

  const shopNowLink = screen.getByRole("link", { name: /shop now/i });
  expect(shopNowLink).toHaveAttribute("href", "/store");
});

it("should render featured items", async () => {
  const items = mockData.data.products.edges;
  render(
    <MemoryRouter>
      <Home items={items} />
    </MemoryRouter>
  );
  expect(await screen.findByText(/slides/i)).toBeInTheDocument();
  expect(await screen.findByText(/sweatpants/i)).toBeInTheDocument();
});

it("should render home and match snapshot", () => {
  const items = mockData.data.products.edges;

  const { container } = render(
    <MemoryRouter>
      <Home items={items} />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
