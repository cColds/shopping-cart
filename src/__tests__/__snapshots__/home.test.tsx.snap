// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders home component and matches snapshot 1`] = `
<div>
  <section
    class="flex max-w-4xl flex-col items-center gap-5 p-12"
  >
    <h1
      class="text-center"
    >
      Your one-stop destination for all your shopping needs
    </h1>
    <p
      class="text-center text-lg"
    >
      Discover an unparalleled shopping experience with our extensive selection of products, unbeatable prices, and exceptional customer service. Shop now and transform your shopping journey with us.
    </p>
    <a
      class="w-[150px] overflow-hidden truncate rounded-lg bg-primary-color p-3 text-center font-bold text-white"
      href="/store"
    >
      Shop now
    </a>
  </section>
  <section
    class="m-12 flex w-full flex-col items-center bg-slate-50 p-8 shadow-lg"
  >
    <h2
      class="text-center"
    >
      Featured Items
    </h2>
    <ul
      class="m-8 flex w-full flex-wrap justify-center gap-12 p-8"
    >
      <li
        class="flex cursor-pointer flex-col items-center"
      >
        <a
          aria-label="Slides for $25"
          class="w-[200px]"
          href="/store/slides"
        >
          <img
            alt="Slides"
            class="rounded-tl-lg rounded-tr-lg"
            draggable="false"
            height="200px"
            src="https://cdn.shopify.com/s/files/1/0688/1755/1382/products/slides.jpg?v=1675447358"
            width="200px"
          />
          <div
            class="flex h-24 flex-col gap-1 rounded-bl-lg rounded-br-lg bg-white p-2.5 shadow-lg"
          >
            <h3
              class="truncate"
            >
              Slides
            </h3>
            <p
              class="truncate"
            >
              $
              25
            </p>
          </div>
        </a>
      </li>
      <li
        class="flex cursor-pointer flex-col items-center"
      >
        <a
          aria-label="Sweatpants for $35"
          class="w-[200px]"
          href="/store/sweatpants"
        >
          <img
            alt="Sweatpants"
            class="rounded-tl-lg rounded-tr-lg"
            draggable="false"
            height="200px"
            src="https://cdn.shopify.com/s/files/1/0688/1755/1382/products/GreenSweatpants01.jpg?v=1675455387"
            width="200px"
          />
          <div
            class="flex h-24 flex-col gap-1 rounded-bl-lg rounded-br-lg bg-white p-2.5 shadow-lg"
          >
            <h3
              class="truncate"
            >
              Sweatpants
            </h3>
            <p
              class="truncate"
            >
              $
              35
            </p>
          </div>
        </a>
      </li>
    </ul>
  </section>
</div>
`;
